### PR TITLE
Fetch role path for identity inside JWT

### PR DESCRIPTION
**Pull request by nessiLahav**
### What does this PR do?
Add role path configuration for identity inside JWT.

The PR contains the following logic:
	• URL behavior stay the same 
	         ○ only host are supported
		○ full path of the identity 
		○ if both url and token-app-property configured we throw an error
	• For users inside the token: we have a new identity-path property 
	        ○ only host are supported
		○ if not configured - default behavior is host in root policy
		○ if resource exists but empty - we throw an error
                 ○ the property contains only the path in the policy, the role type 'host' will be added behind the scenes 

For example:
identity-path= apps
identity inside token= 00uksxz3o2vFJCX230h7

The calculated identity is: host/apps/00uksxz3o2vFJCX230h7

Missing:
UTs and status logic

Example to the relevant flow in the logs:
```
CONJ00114D Fetching identity by interface: 'token-identity-provider'
  Sequel::Postgres::Database (0.5ms)  SELECT * FROM "resources" WHERE "resource_id" = 'cucumber:variable:conjur/authn-jwt/okta/token-app-property'
  Sequel::Postgres::Database (0.4ms)  SELECT * FROM "secrets" WHERE ("secrets"."resource_id" = 'cucumber:variable:conjur/authn-jwt/okta/token-app-pr
CONJ00060I Retrieved value 'sub' of resource name 'cucumber:variable:conjur/authn-jwt/okta/token-app-property'
CONJ00061D Checking if field 'sub' is in the token...
CONJ00097D Successfully found field 'sub' in token and its value is '00uksxz3o2vFJCX230h7'
CONJ00110D Fetching identity path...
CONJ00112D Validating identity path configuration...
  Sequel::Postgres::Database (0.5ms)  SELECT * FROM "resources" WHERE "resource_id" = 'cucumber:variable:conjur/authn-jwt/okta/identity-path'
  Sequel::Postgres::Database (0.5ms)  SELECT * FROM "resources" WHERE "resource_id" = 'cucumber:variable:conjur/authn-jwt/okta/identity-path'
  Sequel::Postgres::Database (0.5ms)  SELECT * FROM "secrets" WHERE ("secrets"."resource_id" = 'cucumber:variable:conjur/authn-jwt/okta/identity-pat
CONJ00060I Retrieved value 'apps' of resource name 'cucumber:variable:conjur/authn-jwt/okta/identity-path'
CONJ00113D Identity path configuration validated
CONJ00111I Successfully fetched identity path 'apps'
CONJ00116D Adding identity prefix 'apps' to identity '00uksxz3o2vFJCX230h7'...
CONJ00117D Successfully added identity prefix, calculated identity is: 'apps/00uksxz3o2vFJCX230h7'
CONJ00116D Adding identity prefix 'host' to identity 'apps/00uksxz3o2vFJCX230h7'...
CONJ00117D Successfully added identity prefix, calculated identity is: 'host/apps/00uksxz3o2vFJCX230h7'
CONJ00115I Successfully fetched identity 'host/apps/00uksxz3o2vFJCX230h7' by interface: 'token-identity-provider'
CONJ00102D 'get_jwt_identity' passed successfully
CONJ00098I Successfully found JWT identity 'host/apps/00uksxz3o2vFJCX230h7'
```

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/domain/logs.rb [hunk 1]
```diff
@@ -555,6 +555,51 @@ module AuthnJwt
         code: "CONJ00109D"
       )
 
+      FetchingIdentityPath = ::Util::TrackableLogMessageClass.new(
```
Comment (tzheleznyak) [inline app/domain/logs.rb:558]: @shulifink Can you go over this messages?

CC : @hilagross

File: app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+              LogMessages::Authentication::AuthnJwt::IdentityPathNotConfigured.new(
+                identity_path_resource_id
+              )
+            )
+          end
+          @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedIdentityPathConfiguration.new)
+        end
+
+        def identity_path_resource_exists?
+          return @identity_path_resource_exists unless @identity_path_resource_exists.nil?
```
Comment (tzheleznyak) [inline app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:48]:
```suggestion
          return @identity_path_resource_exists if defined?(@identity_path_resource_exists.nil?)
          @identity_path_resource_exists = !identity_path_resource.nil?
```

Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:48]: @orenbm will be proud of me

**Comment by tzheleznyak** [comment]
Notice that in my PR here im doing many changed in identity_from_decoded_token_provider that might cause you conflicts
https://github.com/cyberark/conjur/pull/2261

File: app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb (hunk 2 above)
Comment (orenbm) [reply app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:48]: ❤️

Comment (nessiLahav) [reply app/domain/authentication/authn_jwt/identity_providers/fetch_identity_path.rb:48]: this is a must? its looks more complicated

File: app/domain/errors.rb [hunk 3]
```diff
@@ -474,6 +474,11 @@ module AuthnJwt
         code: "CONJ00098E"
       )
 
+      MissingIdentityName = ::Util::TrackableErrorClass.new(
```
Comment (nessiLahav) [inline app/domain/errors.rb:477]: @shulifink please review

File: app/domain/logs.rb (hunk 1 above)
Comment (nessiLahav) [reply app/domain/logs.rb:558]: @shulifink i put some examples of the logs in the PR comment

**Pull request closed without merging**